static/report/js/compare.js

```diff
@@ -0,0 +1,637 @@
+  function first_page (resp) {
+    // console.log('resp', resp);
+    payload = resp;
+    do_payload( );
+  }
+
+  function do_payload ( ) {
+    if (opts.callback && opts.callback.call) {
+      opts.callback(payload);
+    } else {
+      // console.log(payload);
+    }
+  }
+
+  function init ( ) {
+    if (query.begin && query.end) {
+      refresh(query.begin, query.end);
+      // do_query(query.begin, query.end, first_page);
+    }
+    page( );
+    return page;
+  }
+
+  page.refresh = refresh;
+
+  return init( );
+}
+
+function time_in_range (data, opts) {
+  var range = {
+    high: 180
+  , low: 80
+  };
+
+  function sgv (d) { return d.sgv; }
+  function is_clean (d) { return sgv(d) > 39; }
+  function is_in_range (d) { return sgv(d) >= range.low && sgv(d) <= range.high; }
+  function is_high (d) { return sgv(d) > range.high; }
+  function is_low (d) { return sgv(d) < range.low; }
+  var results = [ ];
+  var days = d3.nest( )
+      .key(function (d) {
+        return  Date.create(d.dateString).format('{yyyy}-{MM}-{dd}')
+      } )
+      .rollup(function (leaves) {
+        var clean = leaves.filter(is_clean);
+        var nominal = clean.filter(is_in_range);
+        var highs = clean.filter(is_high);
+        var lows = clean.filter(is_low);
+        return {
+          length: clean.length
+        , color: opts.color || '#eee'
+        , target: nominal.length / clean.length * 100
+        , highs: highs.length / clean.length * 100
+        , lows: lows.length / clean.length * 100
+        , mean: d3.mean(clean, sgv)
+        , median: d3.median(clean, sgv)
+        };
+      } )
+    .entries(data);
+    ;
+  return days;
+}
+
+function ranger (dom, opts) {
+  var scales = { x: { }, y: { } };
+  var dimensions = {height: null, width: null };
+  var margin = {top: 20, right: 50, bottom: 20, left: 50};
+  var xAxis, yAxis;
+  var dom_width, dom_height;
+  var width, height;
+  var root = { };
+  var dots = { };
+  var chart = { };
+
+  function my ( ) {
+    // dots = chart.selectAll('circle')
+    // dots.enter( )
+      // .append("circle")
+    // console.log('data', my.data);
+    var selection = dots.selectAll('.dots').data(my.data || [ ]);
+    // console.log('dots', dots, selection);
+    selection.enter( ).append('circle').attr('class', 'dots');
+    selection.exit( ).remove( );
+    selection.call(render_circles)
+    // .exit( ).remove( );
+  }
+
+  function make_x_axis ( ) {
+    return d3.svg.axis( )
+      .scale(scales.x)
+      ;
+  }
+
+  function make_y_axis ( ) {
+    return d3.svg.axis()
+      .scale(scales.y)
+      ;
+  }
+
+  function gridlines (selection) {
+    selection.selectAll('.x.gridlines')
+      // .attr("transform", "translate(0," + height + ")")
+      .attr("transform", "translate(" + margin.left + "," + height + ")")
+      // .attr("transform", "translate(" + margin.left + ", " + (margin.top) + ")")
+      .call(make_x_axis( )
+        .tickSize(-height, 0, 0)
+        .tickFormat("")
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2165148169/issues/source?bid=2452379&fileBranchId=2967311#l484)